### ISSUE TITLE
Request permission to use JColor library

### ISSUE THREAD

**Issue by dojh111** [nus-cs2113-AY2021S1/forum#86]
## Library

[JColor](https://github.com/dialex/JColor)

## Purpose

We intend to use this library to see if it will help with allowing colours for text outputs in the windows CMD terminal. If it does not work, we will be removing it from the project.

## License

MIT License - https://github.com/dialex/JColor/blob/main/LICENSE

**Comment by okkhoy** [comment]
Approved 👍

**Comment by okkhoy** [comment]
reopening as a reference to others.

**Comment by Chongjx** [comment]
@dojh111 I am trying to use this library as well but apparently windows terminal does not support ansi escape codes. So the output became [0m(string)[0m instead. I am wondering if its working for you.

**Comment by dojh111** [comment]
@Chongjx I did some research and I think the issue is the windows command terminal (cmd) just does not enable ANSI escape codes by default. As such I think the only way to get coloured outputs in windows cmd would be to get the user to enable it on their terminal window, or ask them to use powershell instead

**Comment by Chongjx** [comment]
Yeah, I found the reg command for enabling colored output but I'm trying to leave it as the last resort. I tried running with powershell but the output is similar as the command prompt.

**Comment by dojh111** [comment]
I'll continue taking a look to find something that can enable colors on windows cmd, and update here if I find any. Don't want to resort to reg commands also.

**Comment by okkhoy** [comment]
You just have to search a bit harder. There is an easy way out for this problem 😄

**Comment by Chongjx** [comment]
After some more searching, I have found a way (not sure if it's the correct one). Calling SetConsoleMode() to enable virtual terminal processing. The sample code in the first link gives the desired output, however when I try to digest and simplify it I could not the same output.

1. https://stackoverflow.com/questions/52767585/how-can-you-use-vt100-escape-codes-in-java-on-windows
2. https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences

**Comment by okkhoy** [comment]
See #93
